### PR TITLE
chore: Remove few obsolete env vars from deployment scripts

### PR DESCRIPTION
**Pull request by msugakov**
## Description

Triggered by
* https://github.com/stackrox/stackrox/pull/5795#discussion_r1294379793
* https://github.com/stackrox/stackrox/pull/5795#discussion_r1294383119

This change removes `COLLECTOR_IMAGE_REPO`, `SCANNER_IMAGE`, `SCANNER_DB_IMAGE` env vars from dev-deployment scripts and e2e tests. These vars are redundant at this point as everything about images to be deployed must be derived from the image flavor.

Unfortunately, race condition images bypass image flavor and so I could not easily eliminate other parts of the setup like `MAIN_IMAGE` env var. This remains for the future.

This change also makes `deploy/k8s|openshift/deploy*.sh` receptive to `ROX_PRODUCT_BRANDING` environment variable so one can deploy `quay.io/rhacs-eng` images with the following command: `ROX_PRODUCT_BRANDING=RHACS_BRANDING deploy/openshift/deploy.sh`.

## Checklist
- [x] Investigated and inspected CI test results
- ~~[ ] Unit test and regression tests added~~ - this is a change to development scripts that are also used in tests. Not adding automated tests as this feels complicated.

Won't do these as this change doesn't impact commercial users:
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

### Here I tell how I validated my change

1. Ran extra `/test` commands for areas which I think might be affected.
2. Did [image-diff](#issuecomment-1697309327) as suggested by Jan.
3. Used `set -x` around roxctl/helm commands to visually check sanity of arguments. (This isn't the ultimate proof but was useful in development.)

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by connorgorman — APPROVED** [review]
Assuming you want to remove the set +x and -x parts

**Comment by msugakov** [comment]
Sensor did not start because of
```
  Warning  FailedScheduling   41s (x7 over 12m)     default-scheduler   0/3 nodes are available: 3 Insufficient cpu. preemption: 0/3 nodes are available: 3 No preemption victims found for incoming pod..
```

Trying re-run

/test gke-race-condition-qa-e2e-tests

**Comment by jschnath** [comment]
I had a look at the failing gke-race-condition-qa-e2e-tests and it looks like they failed due to some missing quota. Should we rerun these to make sure they pass?
`
error getting cluster operators: the server could not find the requested resource (get clusteroperators.config.openshift.io)
ClusterOperators:
	clusteroperators are missing


[must-gather      ] OUT namespace/openshift-must-gather-xvkkk created
[must-gather      ] OUT clusterrolebinding.rbac.authorization.k8s.io/must-gather-s6dkw created
[must-gather      ] OUT namespace/openshift-must-gather-xvkkk deleted
[must-gather      ] OUT clusterrolebinding.rbac.authorization.k8s.io/must-gather-s6dkw deleted


Error running must-gather collection:
    insufficient quota to match these scopes: [{PriorityClass In [system-node-critical system-cluster-critical]}]

Falling back to `oc adm inspect clusteroperators.v1.config.openshift.io` to collect basic cluster information.
error running backup collection: the server doesn't have a resource type "clusteroperators"


Reprinting Cluster State:
When opening a support case, bugzilla, or issue please include the following summary data along with any other requested information:
error getting cluster version: the server could not find the requested resource (get clusterversions.config.openshift.io version)
ClusterID: 
ClusterVersion: Installing "" for <unknown>: <unknown>
error getting cluster operators: the server could not find the requested resource (get clusteroperators.config.openshift.io)
ClusterOperators:
	clusteroperators are missing


Error from server: insufficient quota to match these scopes: [{PriorityClass In [system-node-critical system-cluster-critical]}]`

**Review by jschnath — COMMENTED** [review]
If I understand your changes correctly you are moving the collector and main images to the same default image registry folder, and also removing the scanner image (because scanner is now a part of stackrox/stackrox?). With all the weird flags in our roxctl generate command this makes sense to me. CI passing is a good sign, but I think some manual testing to make sure that the old roxctl command does the same thing as the new one would be good here.

Maybe a way to do it would be spinning up two clusters one with the old command, the other with your changes, and then checking that they are configured identically? Otherwise I could forsee some unintended consequences or deviation from how it acted before.

**Comment by msugakov** [comment]
Following Jan's good suggestion about diffing what's actually deployed, I did tests on the cluster

<details>
<summary>Details</summary>

## Open source flavor

`deploy/openshift/deploy.sh`

On this branch:

```
$ kubectl -n stackrox get deploy,ds -o wide
NAME                                                     READY   UP-TO-DATE   AVAILABLE   AGE     CONTAINERS           IMAGES                                                         SELECTOR
deployment.apps/admission-control                        3/3     3            3           2m19s   admission-control    quay.io/stackrox-io/main:4.2.x-20-gba86a23ed8                  app=admission-control
deployment.apps/central                                  1/1     1            1           3m48s   central              quay.io/stackrox-io/main:4.2.x-20-gba86a23ed8                  app=central
deployment.apps/central-db                               1/1     1            1           3m49s   central-db           quay.io/stackrox-io/central-db:4.2.x-20-gba86a23ed8            app=central-db
deployment.apps/monitoring                               1/1     1            1           3m59s   grafana,prometheus   grafana/grafana:7.5.15,prom/prometheus:v2.34.0                 app=monitoring
deployment.apps/scanner                                  3/3     3            3           3m40s   scanner              quay.io/stackrox-io/scanner:2.31.0                             app=scanner
deployment.apps/scanner-db                               1/1     1            1           3m40s   db                   quay.io/stackrox-io/scanner-db:2.31.0                          app=scanner-db
deployment.apps/sensor                                   1/1     1            1           2m19s   sensor               quay.io/stackrox-io/main:4.2.x-20-gba86a23ed8                  app=sensor
deployment.apps/stackrox-monitoring-kube-state-metrics   0/1     0            0           3m59s   kube-state-metrics   registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0   app.kubernetes.io/instance=stackrox-monitoring,app.kubernetes.io/name=kube-state-metrics

NAME                       DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE     CONTAINERS                            IMAGES                                                                                                                            SELECTOR
daemonset.apps/collector   6         6         6       6            6           <none>          2m19s   collector,compliance,node-inventory   quay.io/stackrox-io/collector:3.16.0-slim,quay.io/stackrox-io/main:4.2.x-20-gba86a23ed8,quay.io/stackrox-io/scanner-slim:2.31.0   service=collector
```

On `master`:

```
$ kubectl -n stackrox get deploy,ds -o wide
NAME                                                     READY   UP-TO-DATE   AVAILABLE   AGE    CONTAINERS           IMAGES                                                         SELECTOR
deployment.apps/admission-control                        3/3     3            3           28s    admission-control    quay.io/stackrox-io/main:4.2.x-27-g5694ada87f                  app=admission-control
deployment.apps/central                                  1/1     1            1           105s   central              quay.io/stackrox-io/main:4.2.x-27-g5694ada87f                  app=central
deployment.apps/central-db                               1/1     1            1           106s   central-db           quay.io/stackrox-io/central-db:4.2.x-27-g5694ada87f            app=central-db
deployment.apps/monitoring                               1/1     1            1           113s   grafana,prometheus   grafana/grafana:7.5.15,prom/prometheus:v2.34.0                 app=monitoring
deployment.apps/scanner                                  1/3     3            0           100s   scanner              quay.io/stackrox-io/scanner:2.31.x-1-ge4892d2dce               app=scanner
deployment.apps/scanner-db                               1/1     1            0           99s    db                   quay.io/stackrox-io/scanner-db:2.31.x-1-ge4892d2dce            app=scanner-db
deployment.apps/sensor                                   1/1     1            0           28s    sensor               quay.io/stackrox-io/main:4.2.x-27-g5694ada87f                  app=sensor
deployment.apps/stackrox-monitoring-kube-state-metrics   0/1     0            0           113s   kube-state-metrics   registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0   app.kubernetes.io/instance=stackrox-monitoring,app.kubernetes.io/name=kube-state-metrics

NAME                       DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE   CONTAINERS                            IMAGES                                                                                                                                          SELECTOR
daemonset.apps/collector   6         6         3       3            3           <none>          28s   collector,compliance,node-inventory   quay.io/stackrox-io/collector:3.16.0-slim,quay.io/stackrox-io/main:4.2.x-27-g5694ada87f,quay.io/stackrox-io/scanner-slim:2.31.x-1-ge4892d2dce   service=collector
```

## Rhacs-eng (development_build) flavor

`ROX_PRODUCT_BRANDING=RHACS_BRANDING deploy/openshift/deploy.sh`

On this branch:

```
$ kubectl -n stackrox get deploy,ds -o wide
NAME                                                     READY   UP-TO-DATE   AVAILABLE   AGE    CONTAINERS           IMAGES                                                         SELECTOR
deployment.apps/admission-control                        3/3     3            3           41s    admission-control    quay.io/rhacs-eng/main:4.2.x-20-gba86a23ed8                    app=admission-control
deployment.apps/central                                  1/1     1            1           118s   central              quay.io/rhacs-eng/main:4.2.x-20-gba86a23ed8                    app=central
deployment.apps/central-db                               1/1     1            1           119s   central-db           quay.io/rhacs-eng/central-db:4.2.x-20-gba86a23ed8              app=central-db
deployment.apps/monitoring                               1/1     1            1           2m7s   grafana,prometheus   grafana/grafana:7.5.15,prom/prometheus:v2.34.0                 app=monitoring
deployment.apps/scanner                                  3/3     3            3           113s   scanner              quay.io/rhacs-eng/scanner:2.31.0                               app=scanner
deployment.apps/scanner-db                               1/1     1            1           113s   db                   quay.io/rhacs-eng/scanner-db:2.31.0                            app=scanner-db
deployment.apps/sensor                                   1/1     1            1           41s    sensor               quay.io/rhacs-eng/main:4.2.x-20-gba86a23ed8                    app=sensor
deployment.apps/stackrox-monitoring-kube-state-metrics   0/1     0            0           2m7s   kube-state-metrics   registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0   app.kubernetes.io/instance=stackrox-monitoring,app.kubernetes.io/name=kube-state-metrics

NAME                       DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE   CONTAINERS                            IMAGES                                                                                                                      SELECTOR
daemonset.apps/collector   6         6         4       5            4           <none>          41s   collector,compliance,node-inventory   quay.io/rhacs-eng/collector:3.16.0-slim,quay.io/rhacs-eng/main:4.2.x-20-gba86a23ed8,quay.io/rhacs-eng/scanner-slim:2.31.0   service=collector
```

On `master`:

```
$ kubectl -n stackrox get deploy,ds -o wide
NAME                                                     READY   UP-TO-DATE   AVAILABLE   AGE   CONTAINERS           IMAGES                                                         SELECTOR
deployment.apps/admission-control                        3/3     3            3           16s   admission-control    quay.io/stackrox-io/main:4.2.x-27-g5694ada87f                  app=admission-control
deployment.apps/central                                  1/1     1            1           83s   central              quay.io/stackrox-io/main:4.2.x-27-g5694ada87f                  app=central
deployment.apps/central-db                               1/1     1            1           84s   central-db           quay.io/stackrox-io/central-db:4.2.x-27-g5694ada87f            app=central-db
deployment.apps/monitoring                               1/1     1            1           91s   grafana,prometheus   grafana/grafana:7.5.15,prom/prometheus:v2.34.0                 app=monitoring
deployment.apps/scanner                                  0/3     3            0           78s   scanner              quay.io/stackrox-io/scanner:2.31.x-1-ge4892d2dce               app=scanner
deployment.apps/scanner-db                               1/1     1            0           78s   db                   quay.io/stackrox-io/scanner-db:2.31.x-1-ge4892d2dce            app=scanner-db
deployment.apps/sensor                                   0/1     1            0           16s   sensor               quay.io/stackrox-io/main:4.2.x-27-g5694ada87f                  app=sensor
deployment.apps/stackrox-monitoring-kube-state-metrics   0/1     0            0           92s   kube-state-metrics   registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0   app.kubernetes.io/instance=stackrox-monitoring,app.kubernetes.io/name=kube-state-metrics

NAME                       DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE   CONTAINERS                            IMAGES                                                                                                                                          SELECTOR
daemonset.apps/collector   6         6         6       2            6           <none>          16s   collector,compliance,node-inventory   quay.io/stackrox-io/collector:3.16.0-slim,quay.io/stackrox-io/main:4.2.x-27-g5694ada87f,quay.io/stackrox-io/scanner-slim:2.31.x-1-ge4892d2dce   service=collector
```

</details>

Couple things stood out:
* Previously there was no `quay.io/rhacs-eng`/`quay.io/stackrox-io` mixup in open source flavor. This is good.
* There's still no mixup, this is also good :-)
* Previously, the deployment script did not recognize image flavor (or branding which informs the flavor). Now it does.

As I see no issues, I'd go ahead with this change (after removing `set +-x`-s).

File: deploy/common/k8sbased.sh [hunk 1]
```diff
@@ -592,18 +586,24 @@ function launch_sensor {
         kubectl create namespace "$sensor_namespace" &>/dev/null || true
         kubectl -n "$sensor_namespace" get secret stackrox &>/dev/null || kubectl -n "$sensor_namespace" create -f - < <("${common_dir}/pull-secret.sh" stackrox docker.io)
       fi
+      set -x
```
[the remark this answers is not in the document]
Comment (msugakov) [reply deploy/common/k8sbased.sh]: Yes. Removed.

**Pull request merged**